src/language-server/diagnostics.ts

```diff
@@ -146,14 +160,16 @@ export default class DiagnosticGenerator extends Handler {
       const warnings = await this.linter.lintPackage();
       this.reportPackageWarnings(warnings);
     } else {
-      const warnings = await this.linter.lint(this.documents.keys().map(
-          uri => this.converter.getWorkspacePathToFile({uri})));
+      const paths =
+          this.documents.keys()
```

Comment:
mini mini mini optimization: keep the documents keys in a constant to reuse it for `paths` and `diagnosticsByUri`